src/core/src/package_managers/ZypperPackageManager.py

```diff
@@ -72,8 +89,7 @@ def invoke_package_manager(self, command):
 
             self.telemetry_writer.write_execution_error(command, code, out)
             error_msg = 'Unexpected return code (' + str(code) + ') from package manager on command: ' + command
-            self.status_handler.add_error_to_status(error_msg, Constants.PatchOperationErrorCodes.PACKAGE_MANAGER_FAILURE)
-            raise Exception(error_msg, "[{0}]".format(Constants.ERROR_ADDED_TO_STATUS))
+            raise Exception(error_msg)
```

Comment:
Why was this changed? This shouldn't be...

Reply:
Line 75 was removed because I felt that the error should only be reported if all the retries failed.

Because line 75 was removed, the status handler needs to update it in the new spot where the exception is raised, so the `ERROR_ ADDED_TO_STATUS` flag is added in the new exception location on the new line 74. If this was kept as-is, it would still report an error even if a retry succeeded.

Reply:
There's more than this code path calling this. It shouldn't be changed for everyone to accommodate one of the callers.

Reply:
I see what you are saying now. Will revert the changes and will look into how I can accommodate it without modifying this part.